ckanext/dcat/harvesters/_json.py

```diff
@@ -272,15 +272,19 @@ def import_stage(self, harvest_object):
                 'SET CONSTRAINTS harvest_object_package_id_fkey DEFERRED')
             model.Session.flush()
 
-            package_id = \
-                p.toolkit.get_action('package_create')(context, package_dict)
-            log.info('Created dataset with id %s', package_id)
-
         elif status == 'change':
             package_dict['id'] = harvest_object.package_id
-            package_id = \
-                p.toolkit.get_action('package_update')(context, package_dict)
-            log.info('Updated dataset with id %s', package_id)
+
+        if status in ['new', 'change']:
+            try:
+                action = 'package_create' if status == 'new' else 'package_update'
+                message_status = 'Created' if status == 'new' else 'Updated'
+
+                package_id = p.toolkit.get_action(action)(context, package_dict)
+                log.info('%s dataset with id %s', message_status, package_id)
+            except p.toolkit.ValidationError, e:
+                self._save_object_error('Validation Error: %s' % str(e), harvest_object, 'Import')
+                return False
```

Comment:
I think checking for a ValidationError is super and helps to get a proper error message. May a [general try/except just like we have in the RDF harvester](https://github.com/ckan/ckanext-dcat/blob/f15cdfa6a8a9df627107e86e7ee2d98928c4a7b6/ckanext/dcat/harvesters/rdf.py#L385-L390) could be added to make sure we always get a proper error message.

Reply:
Generally it's probably better to be more narrow when catching exceptions so that errors are not hidden but seeing that there's existing code in `dcat` that handles it this way I can update the code to make it consistent.

Reply:
Yes to be narrow and specific is the pythonic way. But the problem with harvesters is, that the error is actually hidden if you don't catch it. The daemon might fail, but usually restarted. So to get an error on the UI, an object_error should be saved.

Reply:
Another approach would be to have a catch all in `ckanext-harvest` instead in case the plugins don't handle the more narrow errors. This would put the onus on the plugins to handle errors they are aware of but if all else fails it won't kill the fetch process as it would be caught by the harvest plugin. 

For now I'll update the code for consistency.

Reply:
I've updated the code to include a catch all exception handling but also have left the existing one in place as I think the existing error message is much clearer for the end user to act on than the catch all message which includes a traceback.

Reply:
@metaodi I've updated the code, let me know what you think